### DISCUSSION TITLE
why log_cli is not a command line option?

### DISCUSSION THREAD

**Discussion by etse1288** [pytest-dev/pytest#10148] [Q&A] [answered]
Hi All,
What's the reason `log_cli` is just an ini option but not command line option as well?
https://github.com/pytest-dev/pytest/blob/main/src/_pytest/logging.py#L247
Using `--log-cli-level XXX` will turn live logging on from command line option, but will update the root logger to `XXX`
Thanks

**Comment by twmr** [comment, marked as answer]
Is there a problem for you that the log-level of the root logger is updated? I guess that it is also updated if `log_cli` is set in an ini file. So can you elaborate why you need a new `log-cli` command line flag?

**Reply by etse1288** [reply]
actually, if i set `log_cli=1` in my ini file the log level does not change. I have used `logging.config.dictConfig()` somewhere in my code to control the log levels of root logger and other loggers. I just want users to have the ability to do this while debugging running `pytest` manually. But then I just found out that I can do `pytest -o log_cli=1` to override the ini values. 

I just don't understand why every option in logging model is using `add_option_ini()`, while `log_cli` is using `parser.addini()`. Just want to know the rationale of it, now that i have found a solution. But of course it would be nice to make it available as a command line option.

**Reply by twmr** [reply]
> But then I just found out that I can do pytest -o log_cli=1 to override the ini values.

Yes, I forgot that we have this way of setting ini variables on the cmd line.  

I think the rationale was that we didn't want to make every `ini` option available as a command line option to not pollute the `--help` text with to many options. Actually, I've just seen that we output both, the `ini` options as well as the command line options in the `--help` text 🤔

**Reply by etse1288** [reply]
ok, thanks for the information!
